src/components/sliding-panel/sliding-panel.tsx

```diff
@@ -15,36 +16,78 @@ export interface SlidingPanelProps extends React.Props<any> {
 
 require('./sliding-panel.scss');
 
-export const SlidingPanel = (props: SlidingPanelProps) => (
-    <div className={classNames('sliding-panel', {
-        'sliding-panel--has-header': !!props.header,
-        'sliding-panel--has-footer': !!props.footer,
-        'sliding-panel--is-narrow': props.isNarrow,
-        'sliding-panel--is-middle': props.isMiddle,
-        'sliding-panel--opened': props.isShown,
-        'sliding-panel--no-padding': props.hasNoPadding,
-        'sliding-panel--off-canvas': props.offCanvas,
-    })}>
-        <div className='sliding-panel__wrapper'>
-            <button className='sliding-panel__close' aria-hidden='true' onClick={() => props.onClose && props.onClose()}>
-                <span>
-                    <i className='argo-icon-close' aria-hidden='true'/>
-                </span>
-            </button>
-            {props.header && (
-                <div className={classNames('sliding-panel__header', {'sliding-panel__header--close-btn-right-padding': props.hasCloseButton})}>
-                    {props.header}
+export const SlidingPanel = (props: SlidingPanelProps) => {
+    return (
+        <KeybindingProvider>
+            <RenderSlidingPanel {...props} />
+        </KeybindingProvider>
+    );
+};
+
+const RenderSlidingPanel = (props: SlidingPanelProps) => {
+
+    const {useKeybinding} = React.useContext(KeybindingContext);
+
+    const closeButtonRef = React.useRef(null);
+    const bodyDivRef = React.useRef(null);
+    const panelHeaderDivRef = React.useRef(null);
+    const panelFooterDivRef = React.useRef(null);
+
+    React.useEffect(() => {
+        if (closeButtonRef && closeButtonRef.current) {
+            closeButtonRef.current.focus();
+        }
+    }, []);
+
+    if (props.isShown) {
```

Comment:
We shouldn't call a hook inside an if condition (see https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)

Instead, you can move this condition to inside the `action`

Reply:
Hey @rbreeze , thanks for catching this. I'll move it to the existing if condition inside.

Reply:
Done